sos/policies/__init__.py

```diff
@@ -124,17 +124,22 @@ def get_service_status(self, name):
         """Returns the status for the given service name along with the output
         of the query command
         """
+        _default = {
+            'name': name,
+            'status': 'missing',
+            'output': ''
+        }
+        if name not in self.services:
+            return _default
+        if 'status' in self.services[name]:
+            # service status has been queried before, return existing info
+            return self.services[name]
         svc = self._query_service(name)
         if svc is not None:
-            return {'name': name,
-                    'status': self.parse_query(svc['output']),
-                    'output': svc['output']
-                    }
-        else:
-            return {'name': name,
-                    'status': 'missing',
-                    'output': ''
-                    }
+            self.services[name]['status'] = self.parse_query(svc['output'])
+            self.services[name]['output'] = svc['output']
+            return self.services[name]
```

Comment:
This changes return value type - 'name' is not further returned in the dict. But it makes sense to drop it, as 1) nobody is using it now, and 2) there is no reason to return also input argument.